add Table to Bookings

rebecalsantos opened the pull request:
-Create new table that includes desk tables. Populate table to 6.
- update POST request to accept an optional desk_id.
- add desks to database/backend GET request

rebecalsantos committed:
add Table to Bookings

-Create new table that includes desk tables. Populate table to 6.
- update POST request to accept an optional desk_id.
- add desks to database/backend GET request

gregdyke commented on server/database_operations.js:
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("pg");
+const config = require("../config");
```
I don't think we have a config

gregdyke commented on server/database_operations.js:
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("pg");
```
should probably import db, as that contains a connexion string

gregdyke commented on server/database_operations.js:
```diff
@@ -0,0 +1,25 @@
+  .then(() => process.exit(0));
+
+function queryDb(query) {
+  return pool.query(query).catch((error) => {
```
then here can be db.query

rebecalsantos replied:
I'm not too sure if I understand this

gregdyke replied:
that's fine. With the approach with config.js, you can ignore my comments on this file

gregdyke commented on server/api.js:
```diff
@@ -24,20 +24,26 @@ router.get("/bookings", async (req, res) => {
 
 router.post("/bookings", async function (req, res) {
 	const userName = req.body.name;
+	const tableName = req.body.table;
```
desk (table would be confusing :D)
and I think it can pass in the desk_id

gregdyke commented on server/api.js:
```diff
@@ -24,20 +24,26 @@ router.get("/bookings", async (req, res) => {
 			"SELECT id from desk_user WHERE username=$1",
 			[userName]
 		);
+		const tableResult = await db.query(
```
if we pass in the desk_id, we won'd need this query

gregdyke commented on server/api.js:
```diff
@@ -24,20 +24,26 @@ router.get("/bookings", async (req, res) => {
 		}
 		const userId = userResult.rows[0].id;
-		const bookingResult = await db.query(insertQuery, [userId, bookingDate]);
+		const tableId = tableResult.rows[0].id;
```
(you didn't check your tableResult - but no matter - don't need it)

gregdyke commented on server/recreate_squema.sql:
```diff
@@ -0,0 +1,15 @@
+drop table if exists desk_table;
+drop table if exists desk_user;
+create table desk_user (id SERIAL PRIMARY KEY, username VARCHAR(200) not null);
+create table desk_table (
```
should be called desk, I think

gregdyke commented on server/api.js:
```diff
@@ -13,7 +13,7 @@ router.get("/bookings", async (req, res) => {
 		// convert array with objects having {username, booking_date} into {name,date}
 		const bookings = result.rows.map((row) => {
-			return { id: row.id, name: row.username, date: row.booking_date };
+			return { id: row.id, name: row.username, table: row.desk_table, date: row.booking_date };
```
the query should join against desk, and we should use both the desk_id and the deskname (as per the trello ticket)

gregdyke reviewed (changes requested):
Very nice start! I like that you changed both POST and GET, though I'm not sure you made them quite right :)

gregdyke commented:
(also, needs the change to package.json to add the script, but that could be done in a followup

rebecalsantos committed:
add Table to Bookings

adding config.js
changing confusing names in tables and in files

teomrd commented on server/database_operations.js:
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("db");
+const config = require("./config");
```
I think in here you need

```js
const { config } = require("./config");
```

as you are exported as a named export (_not default export_)

teomrd commented on server/database_operations.js:
```diff
@@ -0,0 +1,25 @@
+const { Pool } = require("db");
```
And that should be

```js
import { Pool } from "pg";
```

As Pool is a Postgres module

gregdyke replied:
ah, that works too. I thought we would just import db from "./db"

teomrd commented on server/recreate_schema.sql:
```diff
@@ -0,0 +1,16 @@
+create table booking (
+    id SERIAL PRIMARY KEY,
+    desk_user_id INT REFERENCES desk_user(id),
+    desk_id INT REFERENCES desk(id)
```
Seems that you are missing a comma "," in here

rebecalsantos committed:
add table to booking

add desks to database/backend GET request
add npm task to recreate the db
update POST request to accept an optional desk_id ( still not working)

rebecalsantos reviewed (commented):
made changes. deskId doesn't save though

gregdyke commented on server/api.js:
```diff
@@ -11,9 +11,9 @@ router.get("/bookings", async (req, res) => {
 	try {
 		const result = await db.query("SELECT booking.*,desk_user.username from booking INNER JOIN desk_user ON booking.desk_user_id=desk_user.id;"
```
I think this query needs editing to also  inner join against desk, in order to get the desk_name

gregdyke commented on server/populate_db.sql:
```diff
@@ -0,0 +1,20 @@
+INSERT INTO booking (desk_user_id, booking_date) VALUES (3,'2022-01-19');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (3,'2022-01-20');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (1,'2022-01-20');
+INSERT INTO booking (desk_user_id, booking_date) VALUES (1,'2022-01-21');
```
This file is missing the new, desk based bookings

gregdyke commented on server/api.js:
```diff
@@ -24,20 +24,27 @@ router.get("/bookings", async (req, res) => {
 
 router.post("/bookings", async function (req, res) {
 	const userName = req.body.name;
+	const deskId = req.body.deskId;
```
desk_id, probably for consistency with the other filed names, and with the objects returned by the GET request (sorry, naming things well is hard, but worth doing :) )

rebecalsantos committed:
addressing comments

fix get query
fix names
add procfile
add config
